ocaml/typing/typecore.ml

```diff
@@ -813,7 +813,11 @@ let actual_mode_cross_left env ty (actual_mode : Env.actual_mode)
 
 (** Mode cross a mode whose monadic fragment is a right mode, and whose comonadic
     fragment is a left mode. *)
-let alloc_mode_cross_to_max_min env ty { monadic; comonadic } =
+let alloc_mode_cross_to_max_min ~loc env ty { monadic; comonadic } =
+  let () = match constrain_type_jkind env ty (Jkind.of_type_jkind (Jkind.Type.Primitive.any ~why:Dummy_jkind)) with
+    | Ok () -> ()
+    | Error err -> raise (Error (loc, env, Not_a_value(err, None)))
+  in
```

Comment:
This one is getting called with a `Dummy_jkind` [ad: adding tops early, I found a misplaced top in this file]. I couldn't track down where it's coming from easily, but you're likely at this point this should be known.

Reply:
Specifically, it's coming from `type_argument` when `may_coerce = None`, with `ty_expected' = Top` a `Dummy_jkind`, but it's unclear to me what code is triggering this (the stdlib fails to build) and which call-site this might be.